Source/Engine/Level/Level.cpp

```diff
@@ -776,6 +779,8 @@ bool LevelImpl::unloadScene(Scene* scene)
     // Force flush deleted objects so we actually delete unloaded scene objects (prevent from reloading their managed objects, etc.)
     ObjectsRemovalService::Flush();
 
+    Physics::EndPlay();
```

Comment:
What's the purpose of this call?

Reply:
This, paired with `BeginPlay`, is used to reset `Physics::Scenes`. I couldn't find another clean way to achieve this, without it `Physics::Scenes` continues to grow between loads and eventually crashes because the scenes hold references to things that don't exist anymore.

@mafiesto4 Do you have a more flax-ish method or suggestion on how to achieve this? If my method was to be kept I had thought about generalising it so that `Begin/EndPlay` was a member of `EngineService` then any service could hook on to these callbacks.

Reply:
Empty Physics Scenes should be left to stay around as they might be used when the next level gets loaded (eg. additional level streamed in or map changed). If game want, it can manage manually the Physics Scene lifetime by removing it with Destroy (you could add unregistering for PhysicsScene in destructor). We can just destroy all scenes on physics system unload.

Reply:
Let me revisit this. I had engine crashes across subsequent play/stop cycles in the editor, IIRC some collision events were persisting across PIE sessions.

Reply:
OK, so it appears that the issue I originally had has been resolved as the solution became more stable and I've been able to remove the Begin/EndPlay additions.

However, this feels leaky in the editor since now the (empty) physics scenes will persist across multiple PIE sessions. If you think it should be the responsibility of the game to clean these up even in PIE sessions then we'll leave it as-is. My feeling is that exiting a PIE session should almost behave like restarting an exported game.

What would your decision be?